public/docs/advanced/truffle-with-metamask.md

```diff
@@ -1,69 +1,91 @@
-# Accessing Smart Contracts in a Browser
+# Truffle and MetaMask: Working with smart contracts in a browser
 
-To begin interacting with our smart contracts in a browser, first make sure they're compiled, deployed and that we're interacting with them via `web3` in our client-side JavaScript. We recommend using the [truffle-contract library](https://github.com/trufflesuite/truffle-contract), as it makes interacting with contracts easier and more robust.
+Before you can interact with smart contracts in a browser, make sure they're compiled, deployed, and that you're interacting with them via `web3` in client-side JavaScript. We recommend using the [truffle-contract library](https://github.com/trufflesuite/truffle-contract), as it makes interacting with contracts easier and more robust.
 
-For more information on these topics, including using `truffle-contract`, check out our [Pet Shop](/tutorials/pet-shop) or [TutorialToken](/tutorials/robust-smart-contracts-with-openzeppelin) Tutorials.
+<p class="alert alert-info">
+<strong>Note</strong>: For more information on these topics, including using `truffle-contract`, check out our [Pet Shop](/tutorials/pet-shop) or [TutorialToken](/tutorials/robust-smart-contracts-with-openzeppelin) tutorials.
+</p>
 
-# What is MetaMask?
+Once you've done the above, you're ready to use MetaMask.
 
-[MetaMask](https://metamask.io/) is the easiest way to start using Dapps in a browser. It is a Chrome extension that connects to the Ethereum network without running a full node on the browser's machine. It can connect to the Ethereum Main Network; Ropsten, Kovan and Rinkeby testnets; or a local chain such as the TestRPC. For development with Truffle this means we can use our Dapp the same way users will interact with it on a live network.
+## What is MetaMask?
 
-## Installing Metamask
+[MetaMask](https://metamask.io/) is the easiest way to interact with dapps in a browser. It is a Chrome extension that connects to an Ethereum network without running a full node on the browser's machine. It can connect to the main Ethereum network, any of the testnets (Ropsten, Kovan, and Rinkeby), or a local blockchain such as the one created by Truffle Develop or the TestRPC.
```

Comment:
Note: Metamask is available for Firefox now too. https://addons.mozilla.org/en-US/firefox/addon/ether-metamask/

Reply:
Ooh, switch TestRPC here with Ganache, and link to Ganache. 😃 

(Let me know if this means you have to rewrite the whole thing....)

Reply:
It does mean a bit of a rewrite, yes, but that's fine.

The focus of the doc as written was Truffle Develop, with a little note at the end on the TestRPC. Should the focus change to be on Ganache, with little notes at the end on Truffle Develop and Ganache CLI? Let me know what you think would be best.

Reply:
IIRC you have examples for Metamask + TestRPC and Metamask + Truffle Develop. Could you add a section for Metamask + Ganache? (and perhaps switch Metamask + TestRPC to Metamask + Ganache CLI)? 

Note that the process of linking Metamask to Ganache and Ganache CLI is very similar, just different values.